### PR TITLE
Feature 21/coop registration

### PR DESCRIPTION
**Pull request by tsehla42**
## Summary of change

added vuelidate library
added register cooperation component
added global validator helper file
added global styles
added cooperation module to store

### CODE REVIEW

File: src/views/RegisterCooperation.vue [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        if (isValid) {
+          this.isEmailValid = true;
+        }
+      });
+    },
+    registerCooperation() {
+      this.v$.$validate().then((isValid) => {
+        if (isValid) {
+          this.isFormValid = true;
+          console.log(this.v$);
```
Comment (maveric-ui) [inline src/views/RegisterCooperation.vue]: remove console

File: src/views/RegisterCooperation.vue [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      },
+    };
+  },
+  watch: {
+    edrpou(newValue) {
+      console.log(newValue);
+      this.count = newValue.length;
+    },
+    isEmailValid(newValue) {
+      console.log(newValue);
```
Comment (maveric-ui) [inline src/views/RegisterCooperation.vue]: remove console

File: src/views/RegisterCooperation.vue [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+      },
+      edrpou: {
+        requiredValidator,
+        edrpouValidator,
+      },
+    };
+  },
+  watch: {
+    edrpou(newValue) {
+      console.log(newValue);
```
Comment (maveric-ui) [inline src/views/RegisterCooperation.vue]: remove console

File: src/styles/style.scss [hunk 4]
```diff
@@ -0,0 +1,34 @@
+@import url('https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,300;0,400;0,500;0,600;0,700;1,400;1,600&family=Open+Sans:ital,wght@0,300;0,400;0,600;0,700;1,400;1,600&display=swap');
```
Comment (maveric-ui) [inline src/styles/style.scss]: better to divide connection of fonts from global styles
maybe into a separate file: _fonts.scss

File: src/styles/index.scss [hunk 5]
```diff
@@ -0,0 +1,4 @@
+@import '~primevue/resources/themes/saga-blue/theme.css';
```
Comment (maveric-ui) [inline src/styles/index.scss]: why does it duplicate in both places (also in main.ts)?

**Review by lexysline — DISMISSED** [review]

File: src/main.ts [hunk 6]
```diff
@@ -2,13 +2,14 @@ import { createApp } from 'vue';
...
-import { HTTP } from './core/api/http-common';
-import App from './App.vue';
+import { HTTP } from '@/core/api/http-common';
 import PrimeVue from 'primevue/config';
+import App from './App.vue';
 
 import 'primevue/resources/themes/tailwind-light/theme.css';
 import 'primevue/resources/primevue.min.css';
 import 'primeicons/primeicons.css';
+import '@/styles/style.scss';
```
Comment (markuwas) [inline src/main.ts]: I tested import in index.scss, works correct for me, check on your side

File: src/views/RegisterCooperation.vue [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+      },
+    };
+  },
+  watch: {
+    edrpou(newValue) {
+      this.count = newValue.length;
+    },
+    isEmailValid(newValue) {
+      if (newValue === true) {
+        this.isEmailRegistered();
```
Comment (markuwas) [inline src/views/RegisterCooperation.vue]: This test method prevented the project from starting

**Review by markuwas — CHANGES_REQUESTED** [review]

**Review by markuwas — APPROVED** [review]
Good!

**Review by lexysline — APPROVED** [review]

**Pull request merged**